Fixes #772 Toggling darkness off prevents ghosts from seeing one another

MrsTonedOne opened the pull request:
Fixes #772 

Abuses client.images to get around the issue with flat see_invisible scaling not allowing for both being able to see ghosts while not seeing darkness layers.

Also adds a new toggle ghost visibility verb for users who want the old functionality.

MrsTonedOne committed:
Fixes #772 Toggling darkness off prevents ghosts from seeing one another.
Abuses client.images to get around the issue with flat see_invisible scaling not allowing for both being able to see ghosts while not seeing darkness layers.

Chocobro commented:
wow that's an old bug

Mandurrrh commented:
ayyyy :+1:

MrsTonedOne committed:
Moves global var ghostimage over to a static var on ghosts

ike709 commented:
finally fixed this

d3athrow commented on code/modules/mob/dead/observer/logout.dm:
```diff
@@ -1,4 +1,5 @@
 /mob/dead/observer/Logout()
+	client.images -= ghostimages
```
If the person is logging out the client goes away anyways, so for what porpoise?

RemieRichards replied:
shutup smartypants

d3athrow replied:
Bawk

RemieRichards replied:
*throws chicken feed*

MrStonedOne replied:
logout is for mob logouts, not client disconnects.

TZK13 commented:
Wait please no I use this "bug" to able to see things without having ghosts in the way, it's useful as fuck for taking screenshots and shit.

MrsTonedOne committed:
fixes runtime when client disconnects while a ghost.

MrsTonedOne committed:
Adds a new ghost verb to toggle ghost visibility

For users who depended on the old buggy version of see darkness.

reworks seedarkness logic to mash well with the new verb, looks nicer too

MrStonedOne commented:
@TZK13 i added a verb to do that before you edited.

Making it so you don't see yourself is a touch harder, I could but it wouldn't fit in to the see ghosts verb, and i don't think its worth adding yet another verb. this at least restored old functionality for those who want it.

Anonus commented:
How? You're a wizard oh my god you're a wizard

TZK13 commented:
Thank you for adding see ghosts as an option, I agree that while it would be nice there's not really a need for another verb just to hide yourself.

xxalpha commented:
Worst bug. :+1: x :100:

Jordie0608 commented:
Mso chaaaaaangelog

MrsTonedOne committed:
Adds blob overmind cursor mob to ghost darkness vision

Thanks @Ricotez

Also tweaks the wording of related verbs, procs, and variables to match

Moves the scope of the ghostimages (now ghost darkness images) var and updatedateallghostimages proc to the global space so other things can access them

Jordie0608 commented:
Changelog was incorporated into #8090

optimumtact commented:
Awful fix
